src/plugins/auto/executable_network.cpp

```diff
@@ -686,37 +687,44 @@ InferenceEngine::Parameter MultiDeviceExecutableNetwork::GetConfig(const std::st
                 lock.unlock();
                 unsigned int optimalBatchSize = 0;
                 unsigned int requests = 0;
-                std::map<std::string, InferenceEngine::Parameter> options;
-                options["MODEL_PTR"] = std::const_pointer_cast<ngraph::Function>(_network.getFunction());
+                bool bThroughputEnabledInPlugin = false;
                 try {
-                    optimalBatchSize = _core->GetMetric(deviceInfo.deviceName,
-                                    METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
-                    LOG_DEBUG("[AUTOPLUGIN]BATCHING:%s:%ld", "optimal batch size", optimalBatchSize);
+                    // for benchmark through AUTO:CPU,GPU
+                    // SetConfig directly set to CPU/GPU in this case
+                    bThroughputEnabledInPlugin =
+                        _core->GetConfig(deviceInfo.deviceName, CONFIG_KEY(PERFORMANCE_HINT)).as<std::string>() == CONFIG_VALUE(THROUGHPUT);
                 } catch (...) {
-                    LOG_DEBUG("[AUTOPLUGIN]BATCHING:%s", "metric OPTIMAL_BATCH_SIZE not supported");
+                    LOG_DEBUG("[AUTOPLUGIN]GetMetric:%s for %s", "PERF_HINT config not supported", deviceInfo.deviceName.c_str());
                 }
-                if (optimalBatchSize > 1) {
-                    const auto& mode = deviceInfo.config.find(CONFIG_KEY(PERFORMANCE_HINT));
+                const auto& mode = deviceInfo.config.find(CONFIG_KEY(PERFORMANCE_HINT));
+                if (bThroughputEnabledInPlugin ||
+                    (mode != deviceInfo.config.end() && mode->second == CONFIG_VALUE(THROUGHPUT))) {
+                    std::map<std::string, InferenceEngine::Parameter> options;
+                    options["MODEL_PTR"] = std::const_pointer_cast<ngraph::Function>(_network.getFunction());
                     try {
-                        // for benchmark through AUTO:CPU,GPU
-                        // SetConfig directly set to CPU/GPU in this case
-                        auto bThroughputEnabledInPlugin =
-                            _core->GetConfig(deviceInfo.deviceName, CONFIG_KEY(PERFORMANCE_HINT)).as<std::string>() == CONFIG_VALUE(THROUGHPUT);
-                        if (bThroughputEnabledInPlugin ||
-                            (mode != deviceInfo.config.end() && mode->second == CONFIG_VALUE(THROUGHPUT))) {
+                        optimalBatchSize = _core->GetMetric(deviceInfo.deviceName,
+                                        METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
+                        LOG_DEBUG("[AUTOPLUGIN]BATCHING:%s:%ld", "optimal batch size", optimalBatchSize);
+                    } catch (...) {
+                        LOG_DEBUG("[AUTOPLUGIN]BATCHING:%s", "metric OPTIMAL_BATCH_SIZE not supported");
+                    }
+                    if (optimalBatchSize > 1) {
+                        // batching is supported with the device
```

Comment:
@myshevts  @wangleis Maxim, we want to use the ALLOW_AUTO_BATCHING key to decide the optimal num request in AUTO, e.g, when user explicitly disable batching through this key, we will expose the general optimal number instead of the optimal number of batching.  currently this config is been erased after used, does it make sense to maintain this config through?
https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/ie_core.cpp#L552